app/queries/my_facilities/blood_pressure_control_query.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+class MyFacilities::BloodPressureControlQuery
+  include QuarterHelper
+  include MonthHelper
+
+  def initialize(period: :quarter,
+                 quarter: quarter(Time.current),
+                 month: Time.current.month,
+                 year: Time.current.year,
+                 facilities: Facility.all)
+    @period = period
+    @month = month
+    @quarter = quarter
+    @year = year
+    @facilities = facilities
+  end
+
+  def cohort_registrations
+    @period == :month ? monthly_registrations : quarterly_registrations
+  end
+
+  def cohort_controlled_bps
+    @period == :month ? monthly_controlled_bps : quarterly_controlled_bps
+  end
+
+  def cohort_uncontrolled_bps
+    @period == :month ? monthly_uncontrolled_bps : quarterly_uncontrolled_bps
+  end
+
+  def all_time_bps
+    LatestBloodPressuresPerPatient
+      .where(bp_facility_id: @facilities)
+  end
+
+  def all_time_controlled_bps
+    all_time_bps
+      .where('bp_recorded_at > ?', Date.current - 90.days)
+      .where('systolic < 140 AND diastolic < 90')
```

Comment:
We were waiting for #759 to be merged. All of this can now be replaced with a scope in the scenic model and use the `BloodPressure` threshold constants. 😎

Reply:
Ugh, beautiful 😎